src/web/mjs/connectors/MangaParkEN.mjs

```diff
@@ -10,43 +9,113 @@ export default class MangaParkEN extends Connector {
         super.label = 'MangaPark';
         this.tags = [ 'manga', 'multi-lingual' ];
         this.url = 'https://mangapark.net';
+        this.apiURL = `${this.url}/apo/`;
+
+        this.requestOptions.headers.set('x-origin', this.url);
+        this.requestOptions.headers.set('x-referer', `${this.url}/`);
         this.requestOptions.headers.set('x-cookie', 'set=h=1;');
     }
 
     async _getMangaFromURI(uri) {
         const request = new Request(uri, this.requestOptions);
-        const queryMangaTitle = /\/title\/\d+/.test(uri.pathname) ? 'main h3' : 'meta[property="og:title"]';
+        const queryMangaTitle = /\/title\/\d+/.test(uri.pathname) ? 'main h3 a' : 'meta[property="og:title"]';
         const data = await this.fetchDOM(request, queryMangaTitle);
-        return new Manga(this, uri.pathname.match(/\/(\d+)\/?/)[1], (data[0].textContent || data[0].content).trim());
+        const id = parseInt(uri.pathname.match(/\/(\d+)\/?/)[1]);
+        const title = (data[0].textContent || data[0].content).trim();
+        return new Manga(this, id, title);
     }
 
     async _getMangas() {
+        try {
+            return await this._getMangasV3();
```

Comment:
There are chances that _getMangasV3 wont trigger an error but just return an empty array. In this case the V5 fallback wont be called. Maybe catching errors in _getMangasV3 and return an empty array in this case to force v5 fallback?

Reply:
Good find! Thanks. Added if block to check array length.